ni/panels/v1/panel_service.proto

```diff
@@ -0,0 +1,146 @@
+  rpc StopPanel(StopPanelRequest) returns (StopPanelResponse);
+
+  // Enumerate the panels available in the system, including information about the state of the panels and what values they have.
+  // Status Codes for errors:
+  rpc EnumeratePanels(EnumeratePanelsRequest) returns (EnumeratePanelsResponse);
+
+  // Get a value for a control on the panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT: 
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  //   - The value identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  // - NOT_FOUND: 
+  //   - The value with the specified identifier was not found.
+  rpc GetValue(GetValueRequest) returns (GetValueResponse);
+
+  // Try to get a value for a control on the panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT: 
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  //   - The value identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  rpc TryGetValue(TryGetValueRequest) returns (TryGetValueResponse);
+
+  // Set a value for a control on the panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT: 
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  //   - The value identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  rpc SetValue(SetValueRequest) returns (SetValueResponse);
+}
+
+message StartPanelRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Configuration for the panel, packed as Any
+  google.protobuf.Any panel_configuration = 2;
+}
+
+message StartPanelResponse {
+  // Location of the panel 
+  string panel_uri = 1; 
+}
+
+message StopPanelRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Reset all storage associated with panel
+  bool reset = 2;
+}
+
+message StopPanelResponse {
+}
+
+message EnumeratePanelsRequest {
+}
+
+message PanelInformation {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Location of the panel
+  string panel_uri = 2;
+
+  // IDs of all of the values associated with the panel
+  repeated string value_ids = 3;
+}
+
+message EnumeratePanelsResponse {
+  // The list of panels available in the system
+  repeated PanelInformation panels = 1;
+}
+
+message GetValueRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Unique ID of value
+  string value_id = 2;
+}
+
+message GetValueResponse {
+  // The value
+  google.protobuf.Any value = 1;
+}
+
+message TryGetValueRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Unique ID of value
+  string value_id = 2;
+}
+
+message TryGetValueResponse {
+  // The value, if it was found
+  optional google.protobuf.Any value = 1;
```

Comment:
What are you trying to accomplish with the `oneof`? In protobuf3, message fields always have [explicit field presence](https://protobuf.dev/programming-guides/field_presence/#presence-proto3), and the optional keyword really doesn't have any impact.

Reply:
We are using optional keyword to indicate that `TryGetValueResponse` may not have a value. With proto3 this is not technically required, but it's a hint to the user that value may not be present.

Reply:
FYI, grpc-labview does not yet implement support for the `optional` keyword. If you care about that, you would either have to do the work to implement it or maybe include a Boolean field as part of the response to indicate success/failure instead. Using clusters by default for nested messages was perhaps a mistake as it kind deviates from explicit field presence rules across other languages where you can usually still tell whether the field was set without adding the `optional` keyword.

Reply:
Removed the `optional` keyword.